microsoft-edge/extensions-chromium/publish/api/using-addons-publish-api.md

```diff
@@ -0,0 +1,220 @@
+
+```console
+> curl \
+-X POST \
+-H "Content-Type: application/x-www-form-urlencoded" \
+-d "client_id={$Client_ID}" \
+-d "scope=https://addons.edge.microsoft.com/.default" \
+-d "client_secret={$Client_Secret}" \
+-d "grant_type=client_credentials" \
+-v \
+https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token
+```
+
+### Sample response
+
+```json
+{
+  "token_type": "Bearer",
+  "expires_in": 3599,
+  "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1uQ19WWmNBVGZNNXBP..."
+}
+```
+
+For more information, navigate to [OAuth 2.0 client credentials flow on the Microsoft identity platform][AzureOAuthGetToken].
+
+
+<!-- ====================================================================== -->
+## Using the Publish API
+
+Once you have an access token, your app can use the Microsoft Edge Add-ons Publish API.  There are endpoints for getting a list of products, updating a product, and publishing a product.<!-- correct list of actions? there wasn't a get-list section below, only in API Ref. -->
+
+> [!NOTE]
+> Currently, there is no API for creating a new product or updating a product's metadata, such as a description.  You must complete these tasks manually in Microsoft Partner Center.
+
+The examples below use the domain `https://addons.edge.microsoft.com/api`, which is a placeholder and needs to be replaced by another example, given that this API is not live yet.
+
+
+<!-- ====================================================================== -->
+## Getting the list of products
+
+<!-- no section in Part 1 of spec -->
+
+API reference: [Get the list of products](addons-publish-api-reference.md#get-the-list-of-products)
+
+
+<!-- ====================================================================== -->
+## Uploading a package to update an existing submission
+
+Use this API to update the package for an add-on.  This API uploads a package to update an existing draft submission of an add-on product.
+
+```rest
+Endpoint: /v1/products/$productID/submissions/draft/package
+Type: PUT
+Header Parameters: $TOKEN: the access token; Content-Type: application/zip
+Body content: the package file to upload
+```
+
+`$productID` is the ID of the Microsoft Edge Add-on that you want to update.  You can get the product ID in either of the following ways:
+
+*  Login to Microsoft Partner Center.  Navigate to **Microsoft Edge > Overview**, and then select the extension you want the product ID for.  The extension overview page opens.  The GUID in the URL is the product ID.
+
+*  Call the `/products` API to get a list of all products and their product IDs.  For more information about the `/products` API, navigate to the [Add-ons Publish API Reference][AddonsPublishAPIRef].
+
+### Sample request
+
+```console
+> curl \
+-H "Authorization: Bearer $TOKEN" \
+-H "Content-Type: application/zip" \
+-X PUT \
+-T $FILE_NAME \
+-v \
+https://api.partnercenter.microsoft.com/edgeaddons/v1/products/$productID/submissions/draft/package
+```
+
```

Comment:
Add this paragraph at line 134: If the request succeeded and the update process began, you receive a 202 Accepted response status code with a Location header. Make 'GET' requests to the URL in the Location header to find out the status of the operation.